### PR TITLE
Hyundai: Car Port for Staria 2023

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with stock system: `c344fd2492c7a9d2|2023-12-09--12-00-43`
- [x] route with openpilot: `c344fd2492c7a9d2|2023-12-11--09-03-23`
- [x] harness Type: [Hyundai K](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)
- [x] architecture: CAN-FD

Thanks to the community 2023 Hyundai Staria owner inahyung (Discord).

### CODE REVIEW

**Comment by sunnyhaibin** [comment]
@sshane Ready for review!

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ class CAR(StrEnum):
   SANTA_FE_PHEV_2022 = "HYUNDAI SANTA FE PlUG-IN HYBRID 2022"
   SONATA = "HYUNDAI SONATA 2020"
   SONATA_LF = "HYUNDAI SONATA 2019"
+  STARIA_4TH_GEN = "HYUNDAI STARIA 4TH GEN"
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:96]: why 4th gen?

File: selfdrive/car/tests/routes.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ class CarTestRoute(NamedTuple):
   CarTestRoute("b3537035ffe6a7d6|2022-10-17--15-23-49", HYUNDAI.KIA_SPORTAGE_HYBRID_5TH_GEN),
   CarTestRoute("5b7c365c50084530|2020-04-15--16-13-24", HYUNDAI.SONATA),
   CarTestRoute("b2a38c712dcf90bd|2020-05-18--18-12-48", HYUNDAI.SONATA_LF),
+  CarTestRoute("c344fd2492c7a9d2|2023-12-11--09-03-23", HYUNDAI.STARIA_4TH_GEN),
```
Comment (sshane) [inline selfdrive/car/tests/routes.py:120]: this route is 2hrs, is it okay to use a shorter one?

File: selfdrive/car/torque_data/override.yaml [hunk 3]
```diff
@@ -64,6 +64,7 @@ HYUNDAI AZERA HYBRID 6TH GEN: [1.8, 1.8, 0.1]
 KIA K8 HYBRID 1ST GEN: [2.5, 2.5, 0.1]
 HYUNDAI CUSTIN 1ST GEN: [2.5, 2.5, 0.1]
 LEXUS GS F 2016: [2.5, 2.5, 0.08]
+HYUNDAI STARIA 4TH GEN: [2.5, 2.5, 0.1]
```
Comment (sshane) [inline selfdrive/car/torque_data/override.yaml:67]:
```suggestion
HYUNDAI STARIA 4TH GEN: [2.0, 2.0, 0.1]
```

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:96]: The codename `US4` stands for 4th gen: https://thekoreancarblog.com/2023/10/14/hyundai-staria-hybrid-spied/#:~:text=Viewing%20the%20images%2C%20it%20could,it%20as%20an%20Hybrid%20vehicle.

image

**Comment by sshane** [comment]
thanks!

**Pull request merged**